src/controller/CHIPDeviceController.cpp

```diff
@@ -450,6 +450,7 @@ CHIP_ERROR DeviceController::GetDevice(NodeId deviceId, Device ** out_device)
             VerifyOrExit(err == CHIP_NO_ERROR, ReleaseDevice(device));
 
             device->Init(GetControllerDeviceInitParams(), mListenPort, mAdminId);
+            ReturnErrorOnFailure(device->LoadSecureSessionParametersIfNeeded());
```

Comment:
This forces session establishment whenever `GetDevice` is called.  This does not seem right, and is the main thing that I don't think can be put off to a followup here.  Not least because session establishment is right now being converted to async, and making `GetDevice` async does not seem right.

The old code ended up doing this indirectly via `SendCommand` calling `SendMessage` on the device and then the device doing session setup.  For commands we do it via `Device::SendCommands`.

For this new setup, I think the best option is to have `{{asCamelCased parent.name false}}Cluster::ReadAttribute{{asCamelCased name false}}` call a method on the device to do a few things:

1. Set up the attribute path.  That is, we would pass in the things the device does not know (endpoint, cluster, field id) and have it do the work in only one place, so we don't have lots of duplicated code that all does the same thing.
2. Do the `LoadSecureSessionParameters`.  Then later we can update device code to do things async here.
3. Call SendReadRequest.